go/vt/vtgate/scatter_conn.go

```diff
@@ -176,6 +177,8 @@ func (stc *ScatterConn) ExecuteMultiShard(
 		shards = append(shards, shard)
 	}
 
+	autocommit := len(shards) == 1 && canCommit && session.AutocommitApproval()
```

Comment:
Naming is hard and annoying, but IMO `canAutocommit` would be clearer than `canCommit` for the flag that we pass down through the various layers.

Reply:
I was actually not referring to the local variable `autocommit` here, but more the `canCommit` parameter that is plumbed down throughout various call sites and other places.

That (IMO) would be clearer if it was `canAutocommit` but it's a minor point.

Reply:
Oh. There is a good reason why that is named 'canCommit'. It's futuristic: if an autocommit yields multiple statements within the same shard, then the final statement can be execute-committed immediately. Another round-trip saved.

If you look at the logic in engine, this meaning is upheld throughout the code. Will we ever use it that way? Only time will tell.

Reply:
We're debating semantics at this point but I would argue even in that case you are still autocommitting...

That saidl I now have actual code to write so this is the last I will speak of this.

Reply:
I see your point now. I'll rename it :).